apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/DataTransformer.java

```diff
@@ -0,0 +1,233 @@
+package gov.cms.bfd.pipeline.rda.grpc.source;
+
+import com.google.common.collect.ImmutableList;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * Stateful, mutable, non-thread safe object to facilitate transformation of data from incoming RDA
+ * API objects into database entity objects. Every copy method validates the field value in an
+ * appropriate manner and then passes the transformed value to a Consumer (usually a setter on the
+ * entity). All data validation failures are tracked in a List. Following the transformation, the
+ * caller can invoke the isSuccessful method to determine if there were any errors. All copy methods
+ * return this instance so that calls can be chained.
+ */
+public class DataTransformer {
+  private final List<ErrorMessage> errors = new ArrayList<>();
+
+  /**
+   * Determines if all of the transoformations were successful.
+   *
+   * @return true if all transofmrations were successful, false otherwise
+   */
+  public boolean isSuccessful() {
+    return errors.isEmpty();
+  }
+
+  /** @return A (possibly empty) list of all of the transformation errors. */
+  public List<ErrorMessage> getErrors() {
+    return ImmutableList.copyOf(errors);
+  }
+
+  /**
+   * Throws an exception if there are any transformation errors. Makes it easy to generate a
+   * standard exception following all of the transformation steps.
+   *
+   * @throws TransformationException if there are any transformation errors
+   * @return this
+   */
+  public DataTransformer throwIfErrorsPresent() throws TransformationException {
+    if (errors.size() > 0) {
+      throw new TransformationException(
+          String.format("failed with %d errors", errors.size()), getErrors());
```

Comment:
Would it help to log the DCN with this value, I'm sure at some point early on our assumptions about the data will be wrong and we'll immediately have to debug an invalid message.  Trying to think through how we can obtain more information about the message without logging any PII.

Reply:
Thanks.  I have left this generic method unchanged but modified `FissClaimTransformer` to throw its own exception with the DCN included in the error message.